### PR TITLE
feat: Conversion Market config

### PR DESCRIPTION
**Pull request by davidterpay**

### CODE REVIEW

File: oracle/config/aggregate_market.go [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+// that will be used to convert the price of the currency pair to a common currency pair.
+//
+// Price aggregation broadly follows the following steps:
+//  1. Fetch prices for each currency pair from the price feeds.
+//  2. Calculate the provider-weighted median price for each currency pair.
+//  3. Convert the price of each currency pair to a common currency pair using the
+//     AggregatedFeeds field. If there are multiple convertable feeds for a given currency
+//     pair, calculate a median price for each final currency pair, weighted by the
+//     weights of the convertable feeds.
+type AggregateMarketConfig struct {
```
Comment (davidterpay) [inline oracle/config/aggregate_market.go:31]: We can add an additional `Weights` field here that defines the relative weight we should associate with each provider.

File: oracle/config/aggregate_market.go [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+	// the conversion market to convert the BTC/USDT price to BTC/USD. These must be
+	// provided in a topologically sorted order that resolve to the same currency pair
+	// defined in the CurrencyPair field.
+	AggregatedFeeds map[string][][]Conversion `mapstructure:"aggregated_feeds" toml:"aggregated_feeds"`
+}
+
+// FeedConfig represents the configurations for a given price feed. Each currency pair
+// will have its own feed configuration.
+type FeedConfig struct {
+	// CurrencyPair is the currency pair that the oracle will fetch prices for.
```
Comment (davidterpay) [inline oracle/config/aggregate_market.go:48]: At a later time we can add an `Minimum` weight field here that signals how many of the provider's must have returned a price - relative to the weight defined above.

File: oracle/config/aggregate_market.go [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+	// with the resolution configurations for each feed.
+	Feeds map[string]FeedConfig `mapstructure:"currency_pairs" toml:"currency_pairs"`
+
+	// AggregatedFeeds is the list of valid markets (i.e. price feeds) that can be
+	// used to convert the price of the currency pair to a common currency pair. For
+	// example, if the oracle receives a price for BTC/USDT and USDT/USD, it can use
+	// the conversion market to convert the BTC/USDT price to BTC/USD. These must be
+	// provided in a topologically sorted order that resolve to the same currency pair
+	// defined in the CurrencyPair field.
+	AggregatedFeeds map[string][][]Conversion `mapstructure:"aggregated_feeds" toml:"aggregated_feeds"`
```
Comment (davidterpay) [inline oracle/config/aggregate_market.go:42]: At a later time, we can associate the relative weight with each set of conversions such that

[[BTC/USD], [BTC/USDT, USDT/USD]] -> BTC/USD

can be individually weighted by importance.

File: config/local/generate_toml.go [hunk 4]
```diff
@@ -37,20 +37,114 @@ var oracleCfgPath = flag.String("oracle-config-path", "oracle.toml", "path to wr
 // as they are made to this file.
 var LocalConfig = config.OracleConfig{
 	// -----------------------------------------------------------	//
-	// --------------------All Currency Pairs---------------------	//
+	// --------------------Agg Market Config---------------------	//
```
Comment (aljo242) [inline config/local/generate_toml.go]:
```suggestion
	// --------------------Aggregate Market Config---------------------	//
```

File: oracle/config/aggregate_market.go (hunk 1 above)
Comment (aljo242) [reply oracle/config/aggregate_market.go:31]: Add linear issue to track?

File: oracle/config/aggregate_market.go [hunk 5]
```diff
@@ -0,0 +1,186 @@
+package config
+
+import (
+	"fmt"
+
+	oracletypes "github.com/skip-mev/slinky/x/oracle/types"
+)
+
+// AggregateMarketConfig represents the market configurations for how currency pairs will
```
Comment (aljo242) [inline oracle/config/aggregate_market.go:9]: Can we include an example here?

BTC/USDT -> USDT/USD -> BTC/USD

File: oracle/config/aggregate_market.go [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+	// CurrencyPair is the feed that will be used in the conversion.
+	CurrencyPair oracletypes.CurrencyPair `mapstructure:"currency_pair" toml:"currency_pair"`
+
+	// Invert is a flag that indicates if the feed should be inverted prior to being used
+	// in the conversion.
+	Invert bool `mapstructure:"invert" toml:"invert"`
+}
+
+// GetCurrencyPairs returns the set of currency pairs in the aggregate market config.
+func (c AggregateMarketConfig) GetCurrencyPairs() []oracletypes.CurrencyPair {
```
Comment (aljo242) [inline oracle/config/aggregate_market.go]:
```suggestion
func (c *AggregateMarketConfig) GetCurrencyPairs() []oracletypes.CurrencyPair {
```

File: oracle/config/aggregate_market.go [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+
+	for _, cpConfig := range c.Feeds {
+		currencyPairs = append(currencyPairs, cpConfig.CurrencyPair)
+	}
+
+	return currencyPairs
+}
+
+// ValidateBasic performs basic validation on the AggregateMarketConfig.
+func (c AggregateMarketConfig) ValidateBasic() error {
```
Comment (aljo242) [inline oracle/config/aggregate_market.go]:
```suggestion
func (c *AggregateMarketConfig) ValidateBasic() error {
```

File: oracle/config/aggregate_market.go [hunk 8]
```diff
@@ -0,0 +1,186 @@
...
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+// ValidateBasic performs basic validation on the FeedConfig.
+func (c FeedConfig) ValidateBasic() error {
```
Comment (aljo242) [inline oracle/config/aggregate_market.go]:
```suggestion
func (c *FeedConfig) ValidateBasic() error {
```

**Review by aljo242 — APPROVED** [review]

**Pull request merged**